File: Python/base/interview/TreeNode.py
"""
Class representing a node in a binary tree.
Author: Andrew Jarombek
Date: 10/1/2019
"""


class TreeNode:
    def __init__(self, val: int, left, right):
        """
        Construct a node in a binary search tree.
        :param val: Integer value that this node holds.
        :param left: The child node on the left side of this node.
        :param right: The child node on the right side of this node.
        """
        self.val = val
        self.left = left
        self.right = right

File: Java/base/practice/2020/tree/TreeNode.java
package tree;

/**
 * Representation of a node within a tree.  Can be used in a binary tree or binary search tree.
 * @author Andrew Jarombek
 * @since 1/8/2020
 */

// Got a bit of a cold skiing, so unfortunately im not racing the 3k tomorrow.  Excited to race next week though!

public class TreeNode<T> {
    public T value;
    public TreeNode<T> left;
    public TreeNode<T> right;

    public TreeNode(T value) {
        this(value, null, null);
    }

    public TreeNode(T value, TreeNode<T> left, TreeNode<T> right) {
        this.value = value;
        this.left = left;
        this.right = right;
    }
}
